fast_response/FastResponseAnalysis.py

```diff
@@ -992,6 +1020,424 @@ def unblind_TS(self, custom_events=None):
+                    time_mask = [self.duration/2., self.centertime],
+                    pixel_scan=[nside, self._pixel_scan_nsigma],
+                    fixed=['nsignal'],
+                    nsignal=np.array([fluxT*conversion for conversion in nsPerFluxArr],dtype=float), 
+
+                    inject=custom_events
+                    )
+            if val is None:
+                val=temp_val
+            else:
+                val=np.hstack((val,temp_val))
+        ras=val["ra"]
+        decs=val["dec"]
+        nss=val["nsignal"]
+    
+        #Placeholder: not required in fixed index analyses, but future improvements could expand this
+        if "gamma" in val.dtype.names:
+            gammas=val["gamma"]
+        
+        TS_spatial_prior_0=val["TS_spatial_prior_0"]
+        idx_max=np.argmax(TS_spatial_prior_0)
+        ramax,decmax=ras[idx_max],decs[idx_max]
+
+        logProbs=TS_spatial_prior_0/2 
+        finalProbs={}
+
+
+        #for each pixel we have lists [ns_1,ns_2,...ns_i], [logP_1,logP_2,...,logP_3]
+        #we want to integrate over ns
+
+        pixel_list_full=list(zip(ras,decs))
+        for pix in set(pixel_list_full):
+            idxes=np.where((ras==pix[0])&(decs==pix[1]))[0]
+            sort_idx=np.argsort(nss[idxes])
+            idxes=idxes[sort_idx]
+            nss_temp=nss[idxes]
+            probs_temp=logProbs[idxes]
+            ns_diff=np.diff(nss_temp)
+            
+            finalProbs[pix]=-np.inf
+            if(len(idxes)!=len(fluxToTest)):
+                print("Not all fluxes have probabilities")
+                print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+                exit()
+            for i in range(1,len(nss_temp)):
+                #add prior*likelihood*dFlux at each value
+                ns_val=(nss_temp[i]+nss_temp[i-1])/2 #
+                deltaNs=nss_temp[i]-nss_temp[i-1] 
+                prior=prior_func(ns_val,self.index,pix[0],pix[1])
+                finalProbs[pix]=logsumexp([finalProbs[pix],logsumexp([probs_temp[i],probs_temp[i-1]])+np.log(prior)+np.log(deltaNs/2)])
+            
+        #Normalize
+        totalLog=logsumexp(list(finalProbs.values())) 
+        finalProbs={k:v-totalLog for k,v in finalProbs.items()}
+        print("N_pixels in posterior",len(finalProbs))
+        #Write to healpy
+        outputProbHPMap=np.zeros(12*nside**2)
+        ras_graph,decs_graph=zip(*list(finalProbs.keys()))
+        outputProbHPMap[hp.ang2pix(nside=nside,theta=np.array(ras_graph)*180/np.pi, phi=np.array(decs_graph)*180/np.pi,lonlat=True)]=np.exp(np.array(list(finalProbs.values())))
+        
+        t2 = time.time()
+        print("finished posterior calc, took {} s".format(t2-t1))
+
+        #set 0s to nan for plotting 
+        outputProbHPMapPlotting=copy.deepcopy(outputProbHPMap)
+        outputProbHPMapPlotting[outputProbHPMapPlotting == 0] = np.nan
+
+        #Add Plotting
```

Comment:
this is already a huge function. can you move the plotting commands to their own function?